### DISCUSSION TITLE
Zones seem to shift relative to walls over time

### DISCUSSION THREAD

**Discussion by ctdenney** [Hypfer/Valetudo#1697] [Q&A / Support] [answered]
I have a Dreame Z10 Pro and I have noticed that, when a mapping pass is first completed, the zones perfectly match the walls and doorways. However, over time, they seem to shift, with narrow areas that are not a zone at all appearing along some walls, example:
![image](https://user-images.githubusercontent.com/29437087/215556477-47a52569-83c6-4dee-8d4a-f787f4b50c1f.png)
and other zones that used to align with doorways shifting past the doorway. It seems like these kinds of things happen after I have to pick up the robot to dislodge something, or for some other reason it has to reposition itself before continuing (I also noticed it happening when I had to clear the dock and didn't put it down exactly in the same place, but that was clearly my fault, I've since marked the exact position where the dock goes). 

Is this normal/expected behavior? If so, I assume the solution is just that periodically I'll have to do a new mapping pass to sort of reset it? If so, is there a way to do a mapping pass that won't reset the zones/zone names? It wasn't a huge deal, but the last time I did it, I also had to go in and fix all my timers etc., so even if it only has to happen every 6 months or so, it's a minor pain.

**Comment by Hypfer** [comment, marked as answer]
Yep that's expected behavior on the currently supported lidar-based dreame robots on their latest firmwares. They actually already greatly improved the situation with the current firmwares. Before those, the map deteriorated at a much faster pace.

To my current knowledge, for most users, it stays at this minor 1-2px drift indefinitely. Only a few individuals claimed that their map continues to deteriorate even with the latest firmwares. Personally, I haven't seen any major issues like that since those updates.

If it continues to get worse in your setup, unless dreame releases another firmware update that might address that issue, periodically doing a new mapping pass would indeed be the way to go

**Reply by ctdenney** [reply]
Thanks. Yeah, mine eventually deteriorated enough that it was going through unnecessary rooms and doing a lot of movement to try and get to parts of zones that had partially shifted into adjoining rooms, etc. but it's entirely possible that the vast majority of that error happened when I moved the dock, and that the normal drifting wasn't much of an issue at all. The drift I shared a picture of is not enough to cause any issues right now and I only noticed since I was sensitive from the earlier problems.  I'll see how it fares over time now that the dock is in a constant position.
